From b4eba42b72e7fae9ef16d2058f63f73643300e16 Mon Sep 17 00:00:00 2001
From: Dridi Boukelmoune <dridi.boukelmoune@gmail.com>
Date: Sun, 11 Feb 2024 23:22:37 +0100
Subject: [PATCH 1/2] Makefile: Build haredoc(1) with LDLINKFLAGS

This variable is part of the configs but was not applied to haredoc.

To avoid this situation where the configuration for harec2 and haredoc
diverges, a shared HARE_BUILD_ENV variable collects all the variables
needed to configure a hare build execution.
---
 Makefile | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 2482be1f..e3b474eb 100644
--- a/Makefile
+++ b/Makefile
@@ -44,19 +44,20 @@ $(BINOUT)/hare: $(OBJS)
 	@printf 'LD\t%s\n' "$@"
 	@$(LD) $(LDLINKFLAGS) -T $(RTSCRIPT) -o $@ $(OBJS)
 
+HARE_BUILD_ENV = HAREPATH=. HAREC="$(HAREC)" QBE="$(QBE)" AS="$(AS)" \
+	LD="$(LD)" HAREFLAGS="$(HAREFLAGS)" HARECFLAGS="$(HARECFLAGS)" \
+	QBEFLAGS="$(QBEFLAGS)" ASFLAGS="$(ASFLAGS)" LDLINKFLAGS="$(LDLINKFLAGS)"
+
 $(BINOUT)/harec2: $(BINOUT)/hare
 	@printf 'HARE\t%s\n' "$@"
-	@env HAREPATH=. HAREC="$(HAREC)" QBE="$(QBE)" AS="$(AS)" LD="$(LD)" \
-		HAREFLAGS="$(HAREFLAGS)" HARECFLAGS="$(HARECFLAGS)" \
-		QBEFLAGS="$(QBEFLAGS)" ASFLAGS="$(ASFLAGS)" \
-		LDLINKFLAGS="$(LDLINKFLAGS)" \
+	@env $(HARE_BUILD_ENV) \
 		$(BINOUT)/hare build $(HARE_DEFINES) -o $(BINOUT)/harec2 cmd/harec
 
 $(BINOUT)/haredoc: $(BINOUT)/hare
 	@mkdir -p $(BINOUT)
 	@printf 'HARE\t%s\n' "$@"
-	@env HAREPATH=. HAREC="$(HAREC)" QBE="$(QBE)" $(BINOUT)/hare build \
-		$(HARE_DEFINES) -o $(BINOUT)/haredoc ./cmd/haredoc
+	@env $(HARE_BUILD_ENV) \
+		$(BINOUT)/hare build $(HARE_DEFINES) -o $(BINOUT)/haredoc ./cmd/haredoc
 
 docs/html: $(BINOUT)/haredoc
 	mkdir -p docs/html
-- 
2.43.0

